### PR TITLE
convert ClientTestCaseView to TestMethodView

### PR DESCRIPTION
**Pull request by pongad**
This commit converts all uses of ClientTestCaseView to TestMethodView.

The generated code is affected in only one place. There's an extra import in Java smoke test, since the shared logic imports request types, which are not used by the rest of the file. Because we use a formatter to throw out unused imports, I don't think this is a serious concern. Please let me know if it is.

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTestTransformer.java [hunk 1]
```diff
@@ -125,73 +120,27 @@ private MockCombinedView generateMockServiceView(
...
+        clientMethodType = ClientMethodType.PagedRequestObjectMethod;
+      }
       testCaseViews.add(
-          createTestCaseView(
-              context.asRequestMethodContext(method),
+          mockServiceTransformer.createTestMethodView(
+              methodContext,
               testNameTable,
-              Collections.<FieldConfig>emptyList()));
+              initCodeTransformer.createRequestInitCodeContext(
```
Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTestTransformer.java]: I would prefer to make this a local var.

File: src/test/java/com/google/api/codegen/testdata/java_smoke_test_library.baseline [hunk 2]
```diff
@@ -21,6 +21,7 @@ import com.google.common.collect.Lists;
 import com.google.example.library.v1.Book;
 import com.google.example.library.v1.Book.Rating;
 import com.google.example.library.v1.BookName;
+import com.google.example.library.v1.UpdateBookRequest;
```
Comment (shinfan) [inline src/test/java/com/google/api/codegen/testdata/java_smoke_test_library.baseline]: Any reason that this import is added but not used in the code?

File: src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTestTransformer.java (hunk 1 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTestTransformer.java]: I would prefer to make this a local var (also other similar occurrences)

File: src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java [hunk 3]
```diff
@@ -278,26 +258,48 @@ private ClientTestFileView createUnitTestFileView(SurfaceTransformerContext cont
...
+                testNameTable,
+                initCodeTransformer.createRequestInitCodeContext(
+                    methodContext,
+                    new SymbolTable(),
+                    methodConfig.getRequiredFieldConfigs(),
+                    InitCodeOutputType.SingleObject,
+                    valueGenerator),
+                ClientMethodType.CallableMethod));
       } else if (methodConfig.isFlattening()) {
+        ClientMethodType clientMethodType;
```
Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java]: Maybe worth moving this logic into methodConfig so we can just call methodConfig.getMethodType() here.

File: src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTestTransformer.java (hunk 1 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTestTransformer.java]: Oh I saw the description. No as long as we know the reason I don't think it is a big concern.

Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTestTransformer.java]: Done

File: src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java (hunk 3 above)
Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java]: I don't think this works. Different languages set this to different values. Maybe a way out is to put this in a table specific to Java. This logic is only used once though, so I think the benefit is rather small.

**Comment by pongad** [comment]
@shinfan PTAL

File: src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java (hunk 3 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java]: Ummm, fair enough. I thought the client method type is shared across languages.

**Comment by shinfan** [comment]
LGTM. Let @garrettjonesgoogle  take a look too.

File: src/main/java/com/google/api/codegen/viewmodel/testing/TestMethodView.java [hunk 4]
```diff
@@ -14,40 +14,78 @@
...
+import com.google.api.codegen.config.GrpcStreamingConfig.GrpcStreamingType;
 import com.google.api.codegen.viewmodel.ClientMethodType;
 import com.google.api.codegen.viewmodel.InitCodeView;
 import com.google.auto.value.AutoValue;
+import java.util.List;
 
 @AutoValue
 public abstract class TestMethodView {
 
+  public abstract String surfaceMethodName();
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/viewmodel/testing/TestMethodView.java]: Use `clientMethodName`

File: src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTestTransformer.java [hunk 5]
```diff
@@ -125,73 +121,26 @@ private MockCombinedView generateMockServiceView(
         .build();
   }
 
-  private List<ClientTestCaseView> createTestCaseViews(SurfaceTransformerContext context) {
-    ArrayList<ClientTestCaseView> testCaseViews = new ArrayList<>();
+  private List<TestMethodView> createTestCaseViews(SurfaceTransformerContext context) {
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTestTransformer.java]: We should be consistent between `TestMethod` and `TestCase`. I personally prefer `TestCase`. So, I would prefer the view to be named `TestCaseView` (or the original `ClientTestCaseView`)

File: src/main/java/com/google/api/codegen/transformer/MockServiceTransformer.java [hunk 6]
```diff
@@ -103,6 +106,57 @@ public MockServiceTransformer(ValueProducer valueProducer) {
...
+  public TestMethodView createTestMethodView(
+      MethodTransformerContext methodContext,
+      SymbolTable testNameTable,
+      InitCodeContext initCodeContext,
+      ClientMethodType clientMethodType) {
+    Method method = methodContext.getMethod();
+    MethodConfig methodConfig = methodContext.getMethodConfig();
+    SurfaceNamer namer = methodContext.getNamer();
+
+    String surfaceMethodName;
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/MockServiceTransformer.java]: Done

File: src/main/java/com/google/api/codegen/viewmodel/testing/TestMethodView.java (hunk 4 above)
Comment (pongad) [inline src/main/java/com/google/api/codegen/viewmodel/testing/TestMethodView.java]: Done

**Comment by pongad** [comment]
@garrettjonesgoogle PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Comment by pongad** [comment]
YAY! Thank you @garrettjonesgoogle

**Pull request merged**